backend/dataall/modules/s3_datasets_shares/db/share_object_repositories.py

```diff
@@ -29,59 +28,133 @@
+        return S3ShareObjectRepository.count_S3_principal_shares(
             session, group_uri, environment.environmentUri, PrincipalType.Group
         )
 
     @staticmethod
     def count_role_resources(session, role_uri):
-        return ShareObjectRepository.count_role_principal_shares(session, role_uri, PrincipalType.ConsumptionRole)
+        return S3ShareObjectRepository.count_S3_role_principal_shares(session, role_uri, PrincipalType.ConsumptionRole)
 
     @staticmethod
     def delete_env(session, environment):
-        ShareObjectRepository.delete_all_share_items(session, environment.environmentUri)
+        S3ShareObjectRepository.delete_all_S3_share_items(session, environment.environmentUri)
 
 
-class ShareObjectRepository:
+class S3ShareObjectRepository:
     @staticmethod
-    def save_and_commit(session, share):
-        session.add(share)
-        session.commit()
-
-    @staticmethod
-    def list_all_active_share_objects(session) -> [ShareObject]:  ## TODO: Already in shares_base
-        return session.query(ShareObject).filter(ShareObject.deleted.is_(None)).all()
+    def count_S3_principal_shares(session, principal_id: str, environment_uri: str, principal_type: PrincipalType):
+        return (
+            session.query(ShareObject)
+            .join(
+                ShareObjectItem,
+                ShareObjectItem.shareUri == ShareObject.shareUri,
+            )
+            .filter(
+                and_(
+                    ShareObject.principalId == principal_id,
+                    ShareObject.principalType == principal_type.value,
+                    ShareObject.environmentUri == environment_uri,
+                    ShareObjectItem.itemType.in_(
+                        [ShareableType.Table.value, ShareableType.S3Bucket.value, ShareableType.StorageLocation.value]
+                    ),
+                )
+            )
+            .count()
+        )
 
     @staticmethod
-    def find_share(session, dataset: DatasetBase, env, principal_id, group_uri) -> ShareObject:
+    def count_S3_role_principal_shares(session, principal_id: str, principal_type: PrincipalType):
         return (
             session.query(ShareObject)
+            .join(
+                ShareObjectItem,
+                ShareObjectItem.shareUri == ShareObject.shareUri,
+            )
             .filter(
                 and_(
-                    ShareObject.datasetUri == dataset.datasetUri,
                     ShareObject.principalId == principal_id,
-                    ShareObject.environmentUri == env.environmentUri,
-                    ShareObject.groupUri == group_uri,
+                    ShareObject.principalType == principal_type.value,
+                    ShareObjectItem.itemType.in_(
+                        [ShareableType.Table.value, ShareableType.S3Bucket.value, ShareableType.StorageLocation.value]
+                    ),
                 )
             )
-            .first()
+            .count()
         )
 
```

Comment:
get_share_item, used in a resolver --> used in api/types was not used in the frontend, so I removed the whole logic